src/Microsoft.AspNet.Mvc.Core/OptionDescriptors/OptionDescriptorExtensions.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Linq;
+using System.Collections.Generic;
+using Microsoft.AspNet.Mvc.OptionDescriptors;
+
+namespace Microsoft.AspNet.Mvc
+{
+    /// <summary>
+    /// Extension methods for collections of option descriptors.
+    /// </summary>
+    public static class OptionDescriptorExtensions
+    {
+        /// <summary>
+        /// Returns the only instance of <typeparamref name="TInstance"/> from a sequence of option descriptors.
```

Comment:
> `InstanceOf()` is supposed to `throw`, and `InstanceOfOrDefault` is that one that returns `null`

@yishaigalatzer why do we need both?

also both methods `throw` when multiple elements in the list are of the given type. is that desired?

generally, my expectation was for `...InstanceOf<TInstance>()` to be shorthand for `...Select(descriptor => descriptor.Instance).OfType<TInstance>().FirstOrDefault()`; ending with `...First()` if `throw`ing a system-provided `InvalidOperationException` was preferred.